bot/exts/help_channels/_channel.py

```diff
@@ -235,9 +234,10 @@ async def maybe_archive_idle_post(post: discord.Thread, scheduler: scheduling.Sc
         await _close_help_post(post, closing_reason)
         return
 
-    if has_task:
+    if post.id in scheduler:
+        # Cancel any existing close task
         scheduler.cancel(post.id)
```

Comment:
Isn't this going to cancel itself or does the scheduler remove the ID before running the task? Do we need an `asyncio.shield`?

Reply:
The [bot-core Scheduler](https://github.com/python-discord/bot-core/blob/main/pydis_core/utils/scheduling.py#L170) already wraps scheduled coros in asyncio.sheild so they can't cancel themselves.

I think we should be good here from how I understand this works.

Reply:
Yes, I believe you are correct.